### DISCUSSION TITLE
Documentation for type versioning?

### DISCUSSION THREAD

**Discussion by L-as** [MinaProtocol/mina#14343] [Q&A] [answered]
Is there any documentation for the [%%versioned ...] stuff, and why is it done? I can make naive guesses, but it isn't entirely clear to me why the old data structures need to exist when I assume you can support the old wire formats by just translating them at reception time.

**Comment by bkase** [comment, marked as answer]
Yeah the versioned macro sets up the boilerplate for translating the models — docs https://github.com/MinaProtocol/mina/tree/develop/src/lib/ppx_version
